pandas/core/arrays/base.py

```diff
@@ -291,8 +292,15 @@ def _from_factorized(cls, values, original):
     # ------------------------------------------------------------------------
     # Must be a Sequence
     # ------------------------------------------------------------------------
+    @overload
+    def __getitem__(self, item: int | np.integer) -> Any:
+        ...
 
-    def __getitem__(self, item: PositionalIndexer) -> ExtensionArray | Any:
+    @overload
+    def __getitem__(self, item: slice | np.ndarray | Sequence[int]) -> ExtensionArray:
+        ...
+
+    def __getitem__(self, item: PositionalIndexer2D) -> ExtensionArray | Any:
```

Comment:
```suggestion
    def __getitem__(
        self: ExtensionArrayT, item: slice | np.ndarray | Sequence[int]
    ) -> ExtensionArrayT:
        ...

    def __getitem__(
        self: ExtensionArrayT, item: PositionalIndexer
    ) -> ExtensionArrayT | Any:
```

and the ignore in pandas/core/arrays/_mixins.py can be removed

Reply:
Also the docstring does not mention sequence. should maybe update for consistency although we shouldn't change the EA base class api, so maybe we should not include Sequence in the allowed types here?

Reply:
I think it is better to change the API.  We always accept a `Sequence[int]` 

> > Thanks @Dr-Irv generally lgtm. issue with api regarding using Sequence that may require some discussion
> 
> Thinking some more, probably best to remove Sequence altogether here.
> 
> A tuple of ints is a Sequence[int] and returns a Scalar on our 1d arrays in some cases and errors in others, so we need to add an extra overload for tuple to make the the annotations correct. This will then likely create LSP issues with the subclasses that can be backed by a 2d array as the return type of and tuple of ints will be inconsistent will the base implementation.

I've changed it to `list[int]` , because we do accept that, and should reflect that in the EA API